### ISSUE TITLE
Enable Bible references to external Bible app

### ISSUE THREAD

**Issue by breviar-sk** [gyim/breviar-ios#47]
Web+Android solution for SK Breviary has option to switch on biblical references to be "live" (open directly Biblical translation):

- Web: Links point to dkc.kbs.sk website containing official Catholic Bible translation
- Android: Links open "Sväté Písmo" app containing official Catholic Bible translation [author of the Breviar & Bible app is the same so integration is really easy]

For HU verzion, on web solution points to http://szentiras.hu/SZIT.

Live URLs are based on HTML sources (simple usage: {r}reference{/r} with few sophisticated features; details are not important for this issue).

(CZ version has already support for live URLs but not yet decided what to use because there is nothing as 'full catholic transation'.)

Web examples:
SK: https://lh.kbs.sk/cgi-bin/l.cgi?qt=pdt&d=22&m=9&r=2017&o0=647&o1=5376&o2=21240&o3=0&o4=0&o5=0&o6=0&p=mrch#SLAVAOTCU
(see URL from "Žalm 51" -> https://dkc.kbs.sk/?in=%C5%BD51)

HU: https://lh.kbs.sk/cgi-bin/l.cgi?qt=pdt&d=22&m=9&r=2017&j=hu&o0=647&o1=5376&o2=29432&o3=0&o4=0&o5=0&o6=0&p=mrch#SLAVAOTCU
(see URL from kantikum Tób 13 -> http://szentiras.hu/SZIT/T%C3%B3b%2013,%2010-15.%2017-19)

In CGI module, we can add also live URLs for iOS version.

Questions for analysis:

- Which Bible app should be supported? (I personally prefer YouVersion = Holy Bible app https://itunes.apple.com/us/app/bible/id282935706?mt=8; for SK it contains Catholic + Ecumenical translation)
- Could it be feasible to generate links which will open in external app?

**Comment by kominak** [comment]
I really like this feature suggestion and I'd love to implement it. It resonates more with me than adding dark color theme to the app :-D

YouVersion supports a special URL scheme `youversion://`. If our app tries to open such a URL and there is YouVersion installed on the device, it will open the app with the proper book/verse opened.
* https://bentsai.wordpress.com/2013/12/18/youversion-bible-url-scheme/
* https://www.geekswithjuniors.com/note/generating-url-scheme-for-youversions-bible-app-using-python.html

The question is how to handle the case when YouVersion is *not* installed. There are several options.

1. our app will provide one more CGI parameter, e.g. `...&youVersionInstalled=1`, and the CGI module will generate proper links; without this parameter, the module will generate links to national Bible websites (dkc.kbs.sk, ...).
2. the app will keep generating links to national Bible websites; our app will intercept these URLs, and if YouVersion is installed, it will transform these URLs to `youversion://` links.

You can guess which one do I prefer ;)

**Comment by breviar-sk** [comment]
So for now CGI module has to correctly generate OSIS book names & handle new parameter (youVersionInstalled). Most probably this parameter will be another bit-component of current options parameters.

**Comment by kominak** [comment]
Yep, exactly. That seems like the simplest options. But first, let me try if this actually works - I'll try opening YouVersion from some temporary test link...  but I think it will work without any issues.

**Comment by kominak** [comment]
I've just tried opening `youversion://bible?reference=JHN.3.16` and indeed it opened the correct verse in the last used translation (in my case it's slovak catholic SSV translation). 

So in iOS we have to:
* check if the youversion app is present and provide a proper parameter to the CGI module
* fix `webView:(UIWebView *)webView shouldStartLoadWithRequest` in `BRWebViewController`
* tapping on that link shouldn't hide top/bototm bars, right? :)

I have to check how this stuff was supposed to work: `event://linkTouchStart`

**Comment by gyim** [comment]
I contacted YouVersion, and unfortunately they are overwhelmed with requests like this. They will contact me when they will have time for adding Catholic versions for Hungarian. Until then we should really go with szentiras.hu.

**Comment by breviar-sk** [comment]
Currently, the hyperlinks do not work YouVersion (Bible) app even if it is installed. Hyperlinks from Bible references do nothing.
Version: 2.1
